### ISSUE TITLE
Segmentation fault in C++ enum class - 1.8.18

### ISSUE THREAD

**Issue by tttapa** [doxygen/doxygen#7709]
With the latest Doxygen 1.8.18 I get a segmentation fault in my code base that includes an enum class  with an enumerator that has the same name as the enum itself.

For example:
```cpp
/// An enumeration.
enum class Wrap : bool {
    Clamp = false,   ///< Clamp
    Wrap = true,     ///< Wrap         <-- invalidates iterators
    Default = Clamp, ///< Default      <--- segfault when processing this member
};
```
Gives a segmentation fault at line 6990 of doxygen.cpp:
https://github.com/doxygen/doxygen/blob/a1b07ad0e92e4526c9ba1711d39f06b58c2a7459/src/doxygen.cpp#L6990

I've stepped though the section in question, and at first sight, the problem seems to be in these lines: 
https://github.com/doxygen/doxygen/blob/a1b07ad0e92e4526c9ba1711d39f06b58c2a7459/src/doxygen.cpp#L7022-L7023

If the member name matches the enum class's name, `mnsd` already contains this name, `mnsd->add(...)` returns a pointer to the existing `mn`, which is the one that's being iterated over.  
On the next line, `mn->push_back` causes reallocation of the vector `mn` that's being iterated over, invalidating the iterators, so on the next iteration, `md` refers to an empty `unique_ptr` whose lifetime ended when the contents of the vector were moved in the previous line. Dereferencing it results in a segfault. 

If the enumerator with the same name as the enum class is the last one, the problem doesn't manifest itself, because in that case it doesn't matter that the iterators are invalidated, as it happens on the last iteration.
```cpp
/// An enumeration.
enum class Wrap : bool {
    Clamp = false,   ///< Clamp
    Default = Clamp, ///< Default
    Wrap = true,     ///< Wrap         <-- invalidates iterators
}; //                                  <-- not an issue
```

For good measure, I'm using Ubuntu 19.10, with GCC version 9.2.1 20191102 (Ubuntu 9.2.1-17ubuntu1~18.04.1) and Doxygen 1.8.18 (a1b07ad0e92e4526c9ba1711d39f06b58c2a7459, built from source).  
I used the default `doxygen -g` doxyfile. 

I've attached a ZIP with the source code and the `doxygen` binary (debug) I used:  
[Doxygen-7709.zip](https://github.com/doxygen/doxygen/files/4495528/Doxygen-7709.zip)

**Comment by albert-github** [comment]
Thanks for the report the problem has already been identified and fixed in the current doxygen master: c0267cafbd0967944de94fbe8f461641a376a561

**Comment by tttapa** [comment]
Ah I see, last time I pulled was a couple of days ago. I couldn't find any issues about it, so I didn't think it would be fixed already. Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
